sale_order_global_stock_route/views/sale_order_views.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 Tecnativa - Carlos Roca -->
+
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <field name="payment_term_id" position="after">
+                <field name="route_id"/>
+            </field>
+            <field name="order_line" position="attributes">
+                <attribute name="context">{'default_route_id': route_id}</attribute>
```

Comment:
Please user fields_view_get to add a context to void to break inheritence chain..
You can see this link https://github.com/odoo/odoo/blob/f228ae0ce5e635a2c4bcd8d1dd85b14e67e85640/odoo/models.py#L1446